model/task_lifecycle.go

```diff
@@ -1725,27 +1725,38 @@ func resetSystemFailedTask(t *task.Task, description string) error {
 		return nil
 	}
 
-	if err := t.MarkSystemFailed(description); err != nil {
-		return errors.Wrap(err, "marking task as system failed")
-	}
-
-	if time.Since(t.ActivatedTime) > task.UnschedulableThreshold {
+	unschedulableTask := time.Since(t.ActivatedTime) > task.UnschedulableThreshold
+	maxExecutionTask := t.Execution >= evergreen.MaxTaskExecution
+	if unschedulableTask || maxExecutionTask {
+		failureDetails := task.GetSystemFailureDetails(description)
 		// If the task has already exceeded the unschedulable threshold, we
 		// don't want to restart it, so just mark it as finished.
+		msg := "task exceeded unschedulable threshold"
+		if maxExecutionTask {
+			msg = "task exceeded max execution threshold"
+		}
+		grip.Warning(message.Fields{
+			"message": fmt.Sprintf("%s, marking as finished", msg),
+			"task":    t.Id,
+		})
 		if t.DisplayOnly {
 			execTasks, err := task.FindAll(db.Query(task.ByIds(t.ExecutionTasks)))
 			if err != nil {
 				return errors.Wrap(err, "finding execution tasks")
 			}
 			for _, execTask := range execTasks {
 				if !evergreen.IsFinishedTaskStatus(execTask.Status) {
-					if err := MarkEnd(&execTask, evergreen.MonitorPackage, time.Now(), &t.Details, false); err != nil {
+					if err = MarkEnd(&execTask, evergreen.MonitorPackage, time.Now(), &failureDetails, false); err != nil {
 						return errors.Wrap(err, "marking execution task as ended")
 					}
 				}
 			}
 		}
-		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &t.Details, false))
+		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &failureDetails, false))
```

Comment:
Could we also just pass in systemFailureDetails here (or some sort of system failure) and then not have to hit 1758 at all?

Reply:
Sorry I misread; it looks like we need that for the reset